src/analysis/processing/qgsalgorithmdelaunaytriangulation.cpp

```diff
@@ -0,0 +1,174 @@
+/***************************************************************************
+                         qgsalgorithmdelaunaytriangulation.cpp
+                         ---------------------
+    begin                : July 2023
+    copyright            : (C) 2023 by Alexander Bruy
+    email                : alexander dot bruy at gmail dot com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsalgorithmdelaunaytriangulation.h"
+#include "qgsspatialindex.h"
+#include "qgsmultipoint.h"
+#include "qgsattributes.h"
+
+///@cond PRIVATE
+
+QString QgsDelaunayTriangulationAlgorithm::name() const
+{
+  return QStringLiteral( "delaunaytriangulation" );
+}
+
+QString QgsDelaunayTriangulationAlgorithm::displayName() const
+{
+  return QObject::tr( "Delaunay triangulation" );
+}
+
+QStringList QgsDelaunayTriangulationAlgorithm::tags() const
+{
+  return QObject::tr( "delaunay,triangulation,polygons,voronoi" ).split( ',' );
+}
+
+QString QgsDelaunayTriangulationAlgorithm::group() const
+{
+  return QObject::tr( "Vector geometry" );
+}
+
+QString QgsDelaunayTriangulationAlgorithm::groupId() const
+{
+  return QStringLiteral( "vectorgeometry" );
+}
+
+QString QgsDelaunayTriangulationAlgorithm::shortHelpString() const
+{
+  return QObject::tr( "Creates a polygon layer with the Delaunay triangulation corresponding to a points layer." );
+}
+
+QgsDelaunayTriangulationAlgorithm *QgsDelaunayTriangulationAlgorithm::createInstance() const
+{
+  return new QgsDelaunayTriangulationAlgorithm();
+}
+
+void QgsDelaunayTriangulationAlgorithm::initAlgorithm( const QVariantMap & )
+{
+  addParameter( new QgsProcessingParameterFeatureSource( QStringLiteral( "INPUT" ), QObject::tr( "Input layer" ), QList< int >() << QgsProcessing::TypeVectorPoint ) );
+  std::unique_ptr<QgsProcessingParameterNumber> toleranceParam = std::make_unique<QgsProcessingParameterNumber>( QStringLiteral( "TOLERANCE" ), QObject::tr( "Tolerance" ), QgsProcessingParameterNumber::Double, 0, true, 0 );
+  toleranceParam->setHelp( QObject::tr( "Specifies an optional snapping tolerance which can be used to improve the robustness of the triangulation" ) );
```

Comment:
Wow! I didn't know you could do this! Why is it only used on 6 algorithms? This should be everywhere!

Reply:
It's relatively new, introduced after the bulk of the algorithms were already made

Reply:
But yes, should be everywhere 😁